internal/command/deploy/strategy_bluegreen.go

```diff
@@ -507,18 +508,37 @@ func (bg *blueGreen) DestroyBlueMachines(ctx context.Context) error {
 	ctx, span := tracing.GetTracer().Start(ctx, "destroy_blue_machines")
 	defer span.End()
 
+	p := pool.New().
+		WithErrors().
+		WithFirstError().
+		WithMaxGoroutines(16)
+
+	var mu sync.Mutex
 	for _, gm := range bg.blueMachines {
-		if bg.isAborted() {
-			return ErrAborted
-		}
-		err := gm.leasableMachine.Destroy(ctx, true)
-		if err != nil {
-			bg.hangingBlueMachines = append(bg.hangingBlueMachines, gm.launchInput.ID)
-			continue
-		}
+		p.Go(func() error {
+			if bg.isAborted() {
+				return ErrAborted
+			}
+
+			err := gm.leasableMachine.Destroy(ctx, true)
 
-		fmt.Fprintf(bg.io.ErrOut, "  Machine %s destroyed\n", bg.colorize.Bold(gm.leasableMachine.FormattedMachineId()))
+			mu.Lock()
```

Comment:
isn't the mutex only needed to protect`bg.hangingBlueMachines`?

Reply:
It's used for `ErrOut` too as I don't believe stderr isn't safe for concurrent writing.